webpack.config.babel.js

```diff
@@ -1,156 +1,269 @@
-        minify: {
-          removeComments: true,
-        },
-      }),
-      new OpenBrowserPlugin({
-        url: 'http://localhost:8000',
-        // 这里写要打开的浏览器名字，若不填，会打开默认浏览器
-        // Mac系统下可以选：Safari, Google Chrome, Firefox
-        // browser: 'Google Chrome',
-      }),
-    ],
-  });
+  };
 }
-
-if (TARGET === 'build') {
-  module.exports = merge(common, {
+if(target_env === 'prod') { // prod
+  envConfig = {
+    /* ================================================================ */
+    /* 入口分为app和vendor */
+    /* 参考：http://webpack.github.io/docs/code-splitting.html */
+    /* ================================================================ */
     entry: {
-      app: path.resolve(ROOT_PATH, 'app'),
+      app: CONTENT_PATH,
+      vendor: [
+        'jquery',
+        'react',
+        'react-dom',
+        'react-redux',
+        'react-router',
+        'react-router-redux',
+        'redux',
+        'redux-thunk'
+      ],
     },
+    /* ================================================================ */
+    /* app -> bundle.js */
+    /* ================================================================ */
     output: {
       path: path.resolve(ROOT_PATH, 'build'),
-      filename: '[name].js?[chunkhash]',
-    },
-    devtool: 'source-map',
-    module: {
-      noParse: /validate\.js/,
-      loaders: [{
-        test: /\.jsx?$/,
-        loaders: ['babel'],
-        include: path.resolve(ROOT_PATH, 'app'),
-      }, {
-        test: /\.json$/,
-        loader: 'json-loader',
-      }],
+      filename: 'bundle.js?[chunkhash]',
     },
     plugins: [
+      /* ================================================================ */
+      /* IDEA: 这个要不要配置成路径？ */
+      /* ================================================================ */
       new Clean(['build']),
-      /* important! */
+      /* ================================================================ */
+      /* 把公用的module抽出来放进vendor里 */
+      /* （然后原本bundle.js中的这些来源于vendor的module就抽出来了） */
+      /* ================================================================ */
       new webpack.optimize.CommonsChunkPlugin(
-        '[name].js?[chunkhash]'
+        'vendor',
+        'vendor.bundle.js'
       ),
+      /* ================================================================ */
+      /*  Uglify */
+      /* ================================================================ */
       new webpack.optimize.UglifyJsPlugin({
         compress: {
           warnings: false,
         },
       }),
-      new HtmlwebpackPlugin({
-        title: APP_TITLE,
-        template: 'app/index.tpl',
+      /* ================================================================ */
+      /* 注入环境变量 NODE_ENV 为 production */
+      /* 不用这个的时候，Redux会报错说用的版本不是生产版本 */
+      /* 参考：https://github.com/reactjs/redux/issues/1029 */
+      /* ================================================================ */
+      new webpack.DefinePlugin({
+        'process.env.NODE_ENV': '"production"'
```

Comment:
目的：给App设置process.env.NODE_ENV为production.
做法有这些：
http://stackoverflow.com/questions/9198310/how-to-set-node-env-to-production-development-in-os-x
其中，代码的方式是：
`process.env.NODE_ENV = 'production';`
然后是Webpack官网对DefinePlugin的解释：
> If the value is a string it will be used as a code fragment.
官网原文在这里：https://webpack.github.io/docs/list-of-plugins.html#defineplugin
结果就是：
如果'process.env.NODE_ENV': 'production'，value是'production'，做为string被解析成code fragment，但是没有production这个变量啊= =
没有2层引号的时候：
![image](https://cloud.githubusercontent.com/assets/11568087/15034751/7de09a02-12ab-11e6-9729-b0e464ce849d.png)